src/wallet/wallet.cpp

```diff
@@ -1506,6 +1507,29 @@ int CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, bool fUpdate)
     return ret;
 }
 
+void CWallet::ScanForWalletUTXOs()
+{
+    for (std::unique_ptr<CCoinsViewCursor> pcursor(pcoinsTip->Cursor());
```

Comment:
This doesn't deal with unconfirmed TXOs, but neither does `ScanForWalletTransactions`, which this is just intended to be a lightweight fallback for.

I don't know the wallet code well enough to know how it does deal with unconfirmed TXOs, but I expect that if there are issues tracking them, they would be orthogonal to this change.